CONTRIBUTING.md

```diff
@@ -1,28 +1,100 @@
+community, but also contributes to a wider culture of scientific collaboration.
+
+## Bug Reports and Feature Requests
+
+Before reporting a bug or requesting a feature, use Github's issue search to see if anyone else has already done so. Don't reopen an issue tagged with `wontfix` without getting consensus
+from the AllenSDK team in the comment thread.
+
+### Reporting Bugs 
+If there is no existing issue, create a new one. You should include:
+* A brief, descriptive title
+* A clear description of the problem 
+* If you are reporting a bug, your description should contain the following information:
+    * What you did (preferably the actual code or commands that you ran)
+    * What happened
+    * What you were expecting to happen
+    * How your system is configured (operating system, Python version)
+
+If you are comfortable addressing this issue yourself, take a look at the [guide to contributing code](##contributing-code) below.
+
+### Suggesting Features/Enhancements
+Before suggesting a feature or enhancement, please check existing issues as you may find out
+you don't need to create one. When you create an enhancement suggestion, please include
+as many details as possible in the issue template.
+
+When contributing a new feature to the AllenSDK, the maintenance burden is (by default)
+transferred to the AllenSDK team. This means that the benefit of the contribution must be
+compared against the cost of maintaining the feature. 
+
+When suggesting a feature, consider:
+* Is the change clearly explained and motivated?
+* Would the enhancement be useful for most users?
+* Is this a new feature that can stand alone as a third party project?
+
+### Asking Questions 
 
 * Is your question about the Allen _Software Development Kit_, or about Allen Institute data and tools more generally?
     * If the latter, you should check our [online help](http://help.brain-map.org) or the [Allen Brain Map Community Forum](https://community.brain-map.org). If you can't find what you are looking for with the aforementioned resources, you can submit your question using [this form](http://allins.convio.net/site/PageServer?pagename=send_us_a_message_ai).
 * If you do have an AllenSDK question, first check the [documentation](http://alleninstitute.github.io/AllenSDK/index.html) (including the [examples](http://alleninstitute.github.io/AllenSDK/examples.html) and the [api reference](http://alleninstitute.github.io/AllenSDK/allensdk.html)) to make sure that your question is not already addressed.
     * If you can't find an answer in the documentation, please create an issue on Github.
 
-### Code
+## Contributing Code
+
+If you are able to improve the AllenSDK, send us your pull requests!
+Contributing code yourself can be a great way to include the features you want
+in the AllenSDK.
+
+To contribute code, please follow this list:
+* Read contributing guidelines
```

Comment:
is there a hyperlink to the contributing guidelines? back to this page? should just be clear that this is referring to the page they are currently reading.

Reply:
I'll add a link back to the top.